Selenium.Axe/Selenium.Axe.Test/IntegrationTests.cs

```diff
@@ -1,45 +1,91 @@
 ﻿using FluentAssertions;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
 using OpenQA.Selenium;
 using OpenQA.Selenium.Chrome;
 using OpenQA.Selenium.Firefox;
 using OpenQA.Selenium.Support.UI;
 using System;
 using System.IO;
+using System.Collections.Generic;
 
 namespace Selenium.Axe.Test
 {
     [TestClass]
     [DeploymentItem("integration-test-target.html")]
     [DeploymentItem("chromedriver.exe")]
     [DeploymentItem("geckodriver.exe")]
+    [TestCategory("Integration")]
     public class IntegrationTests
     {
         private IWebDriver _webDriver;
         private WebDriverWait _wait;
         private static readonly string IntegrationTestTargetFile = Path.GetFullPath(@"integration-test-target.html");
         private static readonly string IntegrationTestTargetUrl = new Uri(IntegrationTestTargetFile).AbsoluteUri;
 
+        private const string mainElementSelector = "main";
+
         [TestCleanup]
         public virtual void TearDown()
         {
             _webDriver?.Quit();
             _webDriver?.Dispose();
         }
 
+
         [TestMethod]
-        [TestCategory("Integration")]
         [DataRow("Chrome")]
         [DataRow("Firefox")]
-        public void TestAnalyzeTarget(string browser)
+        public void RunScanOnPage(string browser)
         {
+            var expectedToolOptions = new AxeRunOptions()
+            {
+                Rules = new Dictionary<string, RuleOptions>()
+                {
+                    {"color-contrast", new RuleOptions{ Enabled = false} }
+                },
+                RunOnly = new RunOnlyOptions
+                {
+                    Type = "tag",
+                    Values = new List<string>() { "wcag2a" }
+                }
+            };
+
             this.InitDriver(browser);
-            _webDriver.Navigate().GoToUrl(IntegrationTestTargetUrl);
+            LoadTestPage();
+
+            var builder = new AxeBuilder(_webDriver).WithTags("wcag2a").DisableRules("color-contrast");
+
+            var results = builder.Analyze();
+            results.Should().NotBeNull(nameof(results));
+            JsonConvert.SerializeObject(results.ToolOptions).Should().Be(JsonConvert.SerializeObject(expectedToolOptions));
```

Comment:
Is the SerializeObject on both sides just for the purposes of performing a deep comparison? I think you could probably use `results.ToolOptions.Should().BeEquivalentTo(expectedToolOptions);` if so